newIDE/app/src/UI/Search/UseSearchItem.js

```diff
@@ -16,13 +16,83 @@ export class TagSearchFilter<SearchItem: { tags: Array<string> }>
     this.tags = tags;
   }
 
-  isSatisfiedBy(searchItem: SearchItem): boolean {
-    return (
-      this.tags.size === 0 || searchItem.tags.some(tag => this.tags.has(tag))
-    );
+  getPertinence(searchItem: SearchItem): number {
+    return this.tags.size === 0 ||
+      searchItem.tags.some(tag => this.tags.has(tag))
+      ? 1
+      : 0;
   }
 }
 
+/**
+ * Approximately sort the elements from biggest to smallest.
+ *
+ * It does a quick sort but only on the left side. It means that elements with
+ * a poor pertinence won't be sorted as well as the one with a good pertinence.
+ *
+ * This allows a O(n) complexity in most cases, but still a O(n²) in worst
+ * cases scenario (when all the values are the same).
+ */
+export const partialQuickSort = <Element: any>(
```

Comment:
should this function not mutate the array passed and instead return a new sorted array? 
I'm usually quite wary of functions that don't return anything but have modified one of their parameters magically.

Reply:
It's an operation on an array. I don't find it confusing. `push` is also an array operation and it doesn't return a new array.

Reply:
true! but push is called on the object itself as a method, so it warns a bit more that this will affect the object 😇 
I think it's fine either way 👍